src/item_factory.h

```diff
@@ -238,6 +242,20 @@ class Item_factory
         const std::map<const itype_id, std::unique_ptr<itype>> &get_all_itypes() const {
             return m_templates;
         }
+
+        /** Find all templates matching the UnaryPredicate function */
+        static std::vector<const itype *> find( const std::function<bool( const itype & )> &func ) {
+            std::vector<const itype *> res;
+            if( item_controller ) {
```

Comment:
> Perhaps the check is overly paranoid?

I guess so. `item_controller` is a global variable:
```c++
std::unique_ptr<Item_factory> item_controller( new Item_factory() );
```
It should either be instantiated  before any code can call it from the main cycle, or throw an exception if `new` fails due to lack of free memory.